### PR TITLE
ci: Add SonarQube / SonarCloud configuration

### PR DESCRIPTION
**Pull request by paulgessinger**
This PR attempts to add both SonarCloud monitoring to the `main` branch, as well as enabling PR checks using a similar workflow to the physmon comment.

I tested this as much as possible on my fork and PRs **to** that fork from forks of my fork, but ultimately, we'll just have to try it out to see if it works.

### CODE REVIEW

File: .github/workflows/analysis.yml [hunk 1]
```diff
@@ -50,13 +51,23 @@ jobs:
           -DCMAKE_BUILD_TYPE=Debug
           -DCMAKE_CXX_FLAGS="-Werror -gz -g1"
           -DACTS_BUILD_UNITTESTS=on
+          -DACTS_BUILD_EXAMPLES_UNITTESTS=off
```
Comment (AJPfleger) [inline .github/workflows/analysis.yml:54]:
```suggestion
          -DACTS_BUILD_EXAMPLES_UNITTESTS=OFF
```

File: Tests/UnitTests/CMakeLists.txt [hunk 2]
```diff
@@ -112,7 +112,7 @@ function(add_non_compile_test name src)
 endfunction()
 
 add_subdirectory(Core)
-add_subdirectory_if(Examples ACTS_BUILD_EXAMPLES)
+add_subdirectory_if(Examples ACTS_BUILD_EXAMPLES AND ACTS_BUILD_EXAMPLES_UNITTESTS)
```
Comment (AJPfleger) [inline Tests/UnitTests/CMakeLists.txt:115]: Is this related to SonarCube?

File: sonar-project.properties [hunk 3]
```diff
@@ -0,0 +1,17 @@
+sonar.projectKey=paulgessinger_acts
+sonar.organization=paulgessinger
```
Comment (AJPfleger) [inline sonar-project.properties:2]: Could we also have it run for ACTS as a organisation? I think this requires a new account?

Comment (paulgessinger) [reply sonar-project.properties:2]: I overlooked this, but definitely!

File: Tests/UnitTests/CMakeLists.txt (hunk 2 above)
Comment (paulgessinger) [reply Tests/UnitTests/CMakeLists.txt:115]: I missed that case and noticed working on this PR. So I definitely want it.
I can factor this out into a separate PR, but it feels like a too-small change for me to warrant that.

File: .github/workflows/analysis.yml (hunk 1 above)
Comment (paulgessinger) [reply .github/workflows/analysis.yml:54]: We mix cases in the options everywhere. One line above it's also lower case. I don't think this is crucial.

File: sonar-project.properties (hunk 3 above)
Comment (paulgessinger) [reply sonar-project.properties]: Changed now.

**Review by AJPfleger — APPROVED** [review]
Looks good, nice work. Let's see if it works ヾ(＠＾▽＾＠)ﾉ

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [d97b73c12cbc9ddb7b39091c48b32d9ea3333814]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7614593)!

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40356257)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354909)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354908)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354907)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354906)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354905)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354904)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354903)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354902)||
|:red_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354901)||
|:red_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354900)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354899)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354898)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354897)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354896)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354895)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354894)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354893)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354892)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354891)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354890)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354889)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354888)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354887)||
|:red_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354886)||
|:red_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354885)||
|:red_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354884)||
|:red_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354883)||
|:red_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354882)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354881)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/40354880)||

**Comment by AJPfleger** [comment]
https://github.com/acts-project/acts/actions/runs/9619538966

It triggers but we get a startup failure:

> Invalid workflow file: .github/workflows/sonarcloud.yml#L12
> The workflow is not valid. .github/workflows/sonarcloud.yml (Line: 12, Col: 9): Unexpected symbol: '"acts-project"'. Located at position 75 within expression: github.event.workflow_run.conclusion == 'success' && github.repository == "acts-project"
